internal-api/src/main/java/datadog/trace/bootstrap/config/provider/StableConfigSource.java

```diff
@@ -23,16 +24,24 @@ public final class StableConfigSource extends ConfigProvider.Source {
           StableConfigSource.FLEET_STABLE_CONFIG_PATH, ConfigOrigin.FLEET_STABLE_CONFIG);
 
   private final ConfigOrigin fileOrigin;
-
   private final StableConfig config;
 
-  StableConfigSource(String file, ConfigOrigin origin) {
+  StableConfigSource(String filePath, ConfigOrigin origin) {
     this.fileOrigin = origin;
+    File file = new File(filePath);
+    if (!file.exists()) {
+      this.config = StableConfig.EMPTY;
+      return;
+    }
     StableConfig cfg;
     try {
-      cfg = StableConfigParser.parse(file);
+      log.debug("Stable configuration file found at path: {}", file);
+      cfg = StableConfigParser.parse(filePath);
     } catch (Throwable e) {
-      log.debug("Stable configuration file not readable at specified path: {}", file);
+      log.warn(
+          "Encountered the following exception when attempting to read stable configuration file at path: {}, dropping configs.\n",
```

Comment:
```suggestion
          "Encountered the following exception when attempting to read stable configuration file at path: {}, dropping configs.",
```
we typically avoid new-lines in log messages